### PR TITLE
Feature/ge api to bookshelf

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### What's this PR do?
* Merges @nickhartdev's PR with Greyson's worked fixed up
  * Refactors StoryEdit to be a functional component
  * Moves all state info and relevant methods to StoryEditView
  * Refactors the fetch call to use getData instead of getRandomPrompt
* Greyson's additions: 
  * App holds stories(updated periodically to reflect actual prompts and authors)
  * App holds authors & prompts _(realizing now that some of this data is technically duplicated and used in stories, don't know if that's kosher, but it's definitely getting used)_
  * Bookshelves and Books are responsible for cleaning story data
  * Redirect from an unfinished book doesn't require additional fetching
#### Where should the reviewer start?
**Nick:** StorySetupView
**Greyson:** Bookshelves in Library and Welcome views
#### How should this be manually tested?
**Nick** Click new story, choose a prompt, and start your story! A prompt from the appropriate genre should render if the user chooses to use one.
**Greyson** I broke Nick's thing but Imma fix it. 
#### Any background context?
I'll revisit this to take care of ticket #35. The original plan was to write methods to utilize the fact that App holds all prompts, but I ran out of time so the original fetch call is still there.
#### What are the relevant tickets?
None yet!
#### Screenshots (if appropriate)
cat

### CODE REVIEW

**Review by nickhartdev — APPROVED** [review]
Thanks for getting all of these conflicts resolved! Looks good

File: src/ApiHelper/ApiHelper.js [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-    const prompt = await response.json()
-    return prompt
-  }
-
-  static getData = () => {
-    return Promise.resolve(testData.stories)
+  static getData = (type, id) => {
+    return fetch(`${apiHead}/${type}/${id ? id : ''}`)
+      .then(response => response.json())
+      .catch(error => error.message)
```
Comment (Abdeboskey) [inline src/ApiHelper/ApiHelper.js:7]: This method is very cute

File: src/components/Bookshelf/Book.js [hunk 2]
```diff
@@ -18,29 +18,40 @@ const getBookSpine = () => {
   return bookSpines[random];
 };
 
-const Book = ({ story, onClick, popup }) => {
+const findLastWords = (story) => {
+  debugger
+  const lastEntry = story.contributions[story.contributions.length - 1]
+  return `. . . ${lastEntry.substring(5)}`
+} 
+
```
Comment (Abdeboskey) [inline src/components/Bookshelf/Book.js]: I like that

**Review by carlymclift — APPROVED** [review]
Looks good! thanks for working out the conflicts

File: src/components/StoryEditFields/StoryEditFields.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+      {oldStory.prompt !== null && (
+        <h1 id="prompt">PROMPT: {oldStory.prompt.prompt}</h1>
+      )}
+      {prompt !== {} 
+        && <h1 id="prompt">PROMPT: {prompt.prompt}</h1>
       }
+      LAST WORDS: . . .{"testing for now"}
       <input
         id="title"
         type="text"
```
Comment (Abdeboskey) [inline src/components/StoryEditFields/StoryEditFields.js]: Perfect, way to make that happen.

File: src/components/StoryEditView/StoryEditView.js [hunk 4]
```diff
@@ -20,10 +20,12 @@ class StoryEditView extends Component {
...
 
   render() {
+    // console.log(this.props.location.state)
     return (
       <main id="story-edit-section">
         <div id="story-edit-container">
           <StoryEditFields 
+            oldStory={ this.props.location.state.story }
             prompt={ this.props.location.state.prompt }
             updateText={ this.updateText }
```
Comment (Abdeboskey) [inline src/components/StoryEditView/StoryEditView.js:29]: location.state is very useful

**Review by Abdeboskey — APPROVED** [review]
Awesome, way to handle the merges!

**Pull request merged**